admin-dev/themes/new-theme/js/components/grid/extension/link-row-action-extension.js

```diff
@@ -35,6 +35,23 @@ export default class LinkRowActionExtension {
    * @param {Grid} grid
    */
   extend(grid) {
+    $('tr', grid.getContainer()).each(function () {
+      const $parentRow = $(this);
+
+      $('.js-link-row-action[data-clickable-row=1]:first', $parentRow).each(function () {
+        const $rowLink = $(this);
+
+        const $parentCell = $rowLink.closest('td');
+        const clickableCells = $('td.data-type, td.identifier-type, td.badge-type', $parentRow)
+          .not($parentCell)
+        ;
+
+        clickableCells.addClass('cursor-pointer').click(() => {
```

Comment:
What if you click on a checkbox, or a link in the cell?

Reply:
Which is why I only apply the click on some of the cells: `td.data-type, td.identifier-type, td.badge-type` This way toggle cells, checkboxes or actions still work correctly

Reply:
Can you add a little comment to say that?
And are you sure with theses type you're not able to add link? :trollface: 

Other solution could be to get the clicked element and be sure it's a `td` or a `text`.

Reply:
Yes link are added using a LinkColumn to the definition which includes a `link-type` cell
So if you choose not to use the correct type, and override an existing template to include a link where there shouldn't be one... then you probably deserve it!!

I tried to check the event at first but the behaviour was "random" sometimes there's a div, sometimes it's the td which trigger the event, when it's a button you have other use cases Anyway it resulted in overly complex checks that still would allow false positive whereas this method is simpler to understand and (I think) guaranties the expected behaviour (we can always add additional types if need be)

Reply:
Comment added